src/Slot.ts

```diff
@@ -191,13 +203,44 @@ export function connectSlot<T = void, T2 = void>(
         ) => {
             const paramHandlers = handlers[transportKey][param] || []
             const handlerIndex = paramHandlers.indexOf(handler)
-            if (handlerIndex > -1) handlers[transportKey][param].splice(handlerIndex, 1)
-            if (getParamHandlers(param, handlers).length === 0) callLazyDisonnectCallbacks(param)
-            awaitHandlerRegistration(String(transportKey), param)
+
+            if (handlerIndex > -1)
```

Comment:
Nit: You added braces elsewhere, maybe we can add them here too since you've reformatted things a bit